codec-http/src/main/java/io/netty/handler/codec/http/HttpObjectDecoder.java

```diff
@@ -995,25 +1006,43 @@ private final class LineParser extends HeaderParser {
         public AppendableCharSequence parse(ByteBuf buffer) {
             // Suppress a warning because HeaderParser.reset() is supposed to be called
             reset();    // lgtm[java/subtle-inherited-call]
-            return super.parse(buffer);
-        }
-
-        @Override
-        public boolean process(byte value) throws Exception {
+            final int readableBytes = buffer.readableBytes();
+            if (readableBytes == 0) {
+                return null;
+            }
+            final int readerIndex = buffer.readerIndex();
             if (currentState == State.SKIP_CONTROL_CHARS) {
-                char c = (char) (value & 0xFF);
-                if (Character.isISOControl(c) || Character.isWhitespace(c)) {
-                    increaseCount();
-                    return true;
+                final int maxToSkip = Math.min(maxLength, readableBytes);
+                final int firstNonControlIndex = buffer.forEachByte(readerIndex, maxToSkip, SKIP_CONTROL_CHARS_BYTES);
+                if (firstNonControlIndex == -1) {
+                    buffer.skipBytes(maxToSkip);
+                    if (readableBytes > maxLength) {
+                        throw newException(maxLength);
+                    }
+                    seq.reset();
+                    return seq;
                 }
+                // TODO size should be increased as well in order to let super::parse
```

Comment:
`TODO` here is valid: @trustin @normanmaurer 

`size` is used to track the length of body AND read bytes, in order to throw an exception when the `maxLength` is exceeded.
In this PR I'm not increasing `size` with the amount of (already) skipped control chars, meaning that I give the next `parse` (starting from the first non control chars) more room to read body data.

Not sure if it's correct, but giving that I've slightly changed the original behavior and not getting any test error (until now), it means this is uncovered by tests